src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SqlQueryGenerator.cs

```diff
@@ -1036,6 +1089,68 @@ private static bool IsPrimaryKeySort(SearchOptions searchOptions)
             return searchOptions.Sort.All(s => s.searchParameterInfo.Name is SearchParameterNames.ResourceType or SearchParameterNames.LastUpdated);
         }
 
+        private bool IsSortValueNeeded(SearchOptions context)
+        {
+            if (context.Sort.Count == 0)
+            {
+                return false;
+            }
+
+            if (IsPrimaryKeySort(context))
+            {
+                return false;
+            }
+
+            foreach (var searchParamTableExpression in _rootExpression.SearchParamTableExpressions)
+            {
+                if (searchParamTableExpression.Kind == SearchParamTableExpressionKind.Sort ||
+                    searchParamTableExpression.Kind == SearchParamTableExpressionKind.SortWithFilter)
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private static SortContext GetSortRelatedDetails(SearchOptions context)
+        {
+            SortContext sortContext = new SortContext();
+            SearchParameterInfo searchParamInfo = default;
+            if (context.Sort?.Count > 0)
+            {
+                (searchParamInfo, sortContext.SortOrder) = context.Sort[0];
+            }
+
+            sortContext.ContinuationToken = ContinuationToken.FromString(context.ContinuationToken);
+
+            switch (searchParamInfo.Type)
+            {
+                case ValueSets.SearchParamType.Date:
+                    sortContext.SortColumnName = VLatest.DateTimeSearchParam.StartDateTime;
+                    if (sortContext.ContinuationToken != null)
+                    {
+                        DateTime dateSortValue;
+                        if (DateTime.TryParseExact(sortContext.ContinuationToken.SortValue, "o", null, DateTimeStyles.None, out dateSortValue))
+                        {
+                            sortContext.SortValue = dateSortValue;
+                        }
```

Comment:
It looks like `DateTimeStyles.None` converts to DateTime (Local). I think we should use DateTimeOffset if possible, or use `DateTimeStyles.RoundtripKind` that converts to DateTime (Utc). 

https://docs.microsoft.com/en-us/dotnet/api/system.datetime.tryparseexact?view=net-5.0#System_DateTime_TryParseExact_System_String_System_String_System_IFormatProvider_System_Globalization_DateTimeStyles_System_DateTime__